app/models/ead_processor.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+class EadProcessor
+
+  # calls all the methods
+  def self.import_eads(args = {})
+    process_files(args)
+  end
+
+  # sets the url
+  def self.client(args = {})
+    args[:url] || 'https://aspacedev.dlib.indiana.edu/assets/ead_export/'
+  end
+
+  # Open web address with Nokogiri
+  def self.page(args = {})
+    Nokogiri::HTML(open(client(args)))
+  end
+
+  # open file and call extract
+  def self.process_files(args={})
+    for file_link in page(args).css('a')
+      directory = file_link.children.text
+      link = client(args) + file_link.attributes['href'].value
+      next unless should_process_file(args, directory)
+
+      open(link, 'rb') do |file|
+        directory = directory.parameterize.underscore
+        extract_and_index(file, directory)
+      end
+    end
+  end
+
+  # unzip the file and call index
+  def self.extract_and_index(file, directory)
+    Zip::File.open(file) do |zip_file|
+      zip_file.each do |f|
+        path = "./data/#{directory}"
+        FileUtils.mkdir_p path unless File.exist?(path)
+        fpath = File.join(path, f.name)
+        File.delete(fpath) if File.exist?(fpath)
+        zip_file.extract(f, fpath)
+        index_file(fpath, directory)
+      end
+    end
+  end
+
+  # index a file
+  def self.index_file(filename, repository)
+    ENV['REPOSITORY_ID'] = repository
+    ENV['FILE'] = filename
+    solr_url = begin
```

Comment:
Are we just checking that we have a Solr URL here? Doesn't look like the variable is used.

Reply:
not sure, that is copied from the arclight indexer